interface/ceed-vector.c

```diff
@@ -940,27 +940,49 @@ int CeedVectorReciprocal(CeedVector vec) {
 /**
   @brief View a CeedVector
 
-  @param[in] vec    CeedVector to view
-  @param[in] fp_fmt Printing format
-  @param[in] stream Filestream to write to
+  @param[in] vec         CeedVector to view
+  @param[in] num_entries Number of CeedVector entries to view
+  @param[in] fp_fmt      Printing format
+  @param[in] stream      Filestream to write to
 
   @return An error code: 0 - success, otherwise - failure
 
   @ref User
 **/
-int CeedVectorView(CeedVector vec, const char *fp_fmt, FILE *stream) {
+int CeedVectorViewRange(CeedVector vec, CeedSize start, CeedSize stop, CeedSize step, const char *fp_fmt, FILE *stream) {
   const CeedScalar *x;
   CeedCall(CeedVectorGetArrayRead(vec, CEED_MEM_HOST, &x));
 
   char fmt[1024];
   fprintf(stream, "CeedVector length %ld\n", (long)vec->length);
+  if (start != 0 || stop != vec->length || step != 1) {
+    fprintf(stream, "  start: %ld\n  stop:  %ld\n  step:  %ld\n", (long)start, (long)stop, (long)step);
+  }
+  if (stop > vec->length) stop = vec->length;
```

Comment:
Will we allow negative steps? Should we document that one can safely use `INT_MAX`?

Reply:
I added a note that only the subset of the range that's valid for the vector will be printed.

I wasn't thinking of using negative steps but we can add it if you anticipate it being a used.

Reply:
Good point wrt Python's behavior. Adjusted